lib/isexpired.c

```diff
@@ -22,6 +22,24 @@
 
 #ident "$Id$"
 
+/*
+ * sum - adds three non-negative long values
+ *
+ *	the result is capped at LONG_MAX.
+ */
+static long sum(long v1, long v2, long v3)
+{
+	long value;
+
+	if (v1 > LONG_MAX - v2)
+		return LONG_MAX;
+	value = v1 + v2;
+
+	if (value > LONG_MAX - v3)
+		return LONG_MAX;
+
+	return value + v3;
+}
```

Comment:
I'd use something in the name that indicates saturation, even if just an 's'.

Also, an l for long.

For symmetry with __builtin_addcl(), how about addsl()?  Or maybe addl_saturate(), if you prefer a longer name.

Reply:
Renamed to addsl